File: backend/src/main/java/com/ericsson/scrummate/service/AuthService.java
package com.ericsson.scrummate.service;

import com.ericsson.scrummate.dto.AuthResponse;
import com.ericsson.scrummate.dto.LoginRequest;
import com.ericsson.scrummate.dto.RegisterRequest;
import com.ericsson.scrummate.entity.User;
import com.ericsson.scrummate.repository.UserRepository;
import com.ericsson.scrummate.security.JwtUtil;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class AuthService {
    
    @Autowired
    private UserRepository userRepository;
    
    @Autowired
    private PasswordEncoder passwordEncoder;
    
    @Autowired
    private JwtUtil jwtUtil;
    
    public AuthResponse register(RegisterRequest request) {
        if (userRepository.existsByEmail(request.getEmail())) {
            throw new RuntimeException("Email already exists");
        }
        
        User user = new User(request.getEmail(), 
                           passwordEncoder.encode(request.getPassword()), 
                           request.getFullName());
        userRepository.save(user);
        
        String token = jwtUtil.generateToken(user.getEmail());
        return new AuthResponse(token, user.getEmail(), user.getFullName());
    }
    
    public AuthResponse login(LoginRequest request) {
        User user = userRepository.findByEmail(request.getEmail())
                .orElseThrow(() -> new BadCredentialsException("Invalid credentials"));
        
        if (!passwordEncoder.matches(request.getPassword(), user.getPasswordHash())) {
            throw new BadCredentialsException("Invalid credentials");
        }
        
        String token = jwtUtil.generateToken(user.getEmail());
        return new AuthResponse(token, user.getEmail(), user.getFullName());
    }
}

File: frontend/src/services/authService.ts
import axios from 'axios';

const API_BASE_URL = '/api/v1';

interface AuthResponse {
  token: string;
  email: string;
  fullName: string;
}

export const authService = {
  async login(email: string, password: string): Promise<AuthResponse> {
    const response = await axios.post(`${API_BASE_URL}/auth/login`, {
      email,
      password,
    });
    return response.data;
  },

  async register(email: string, password: string, fullName: string): Promise<AuthResponse> {
    const response = await axios.post(`${API_BASE_URL}/auth/register`, {
      email,
      password,
      fullName,
    });
    return response.data;
  },
};
